### ISSUE TITLE
Am getting deadlock error in Thread

### ISSUE THREAD

**Issue by opoudjis** [relaton/relaton#55]
When I try to run rspec on latest code (have bundle updated everything), I am getting the following error on all test cases:

````
  15) Relaton::Db NIST references gets SP
      Failure/Error: require b
      
      ThreadError:
        deadlock; recursive locking
      # /Users/nickn/.rvm/gems/ruby-2.5.1/gems/relaton-gb-0.6.2/lib/relaton_gb.rb:6:in `<top (required)>'
      # ./lib/relaton/registry.rb:25:in `require'
      # ./lib/relaton/registry.rb:25:in `block in register_gems'
      # ./lib/relaton/registry.rb:23:in `each'
      # ./lib/relaton/registry.rb:23:in `register_gems'
      # ./lib/relaton/registry.rb:18:in `initialize'
      # ./lib/relaton/db.rb:12:in `initialize'
      # ./spec/relaton/relaton_spec.rb:9:in `new'
      # ./spec/relaton/relaton_spec.rb:9:in `block (2 levels) in <top (required)>'
````

**Comment by ronaldtse** [comment]
@andrew2net could you help fix this? Thanks.

**Comment by opoudjis** [comment]
Note that this only impacts the relaton gem test cases. relaton-cli is working fine, as are all gems depending on relaton gems. (And thank you for docnumber fix!)

**Comment by andrew2net** [comment]
I implemented new require flavor logic #44. We need to use the latest `relaton-*` gems to make it work correctly.
